bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,134 @@
+from typing import Optional, Union
+
+import arrow
+import discord
+from dateutil import parser
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+# https://discord.com/developers/docs/reference#message-formatting-timestamp-styles
+STYLES = {"Epoch": ("",),
+          "Short Time": ("t", "h:mm A",),
+          "Long Time": ("T", "h:mm:ss A"),
+          "Short Date": ("d", "MM/DD/YYYY"),
+          "Long Date": ("D", "MMMM D, YYYY"),
+          "Short Date/Time": ("f", "MMMM D, YYYY h:mm A"),
+          "Long Date/Time": ("F", "dddd, MMMM D, YYYY h:mm A"),
+          "Relative Time": ("R",)
+          }
+DROPDOWN_TIMEOUT = 60
+
+
+class DateString(commands.Converter):
+    """Convert a relative or absolute date/time string to an arrow.Arrow object."""
+
+    async def convert(self, ctx: commands.Context, argument: str) -> Union[arrow.Arrow, Optional[tuple]]:
+        """
+        Convert a relative or absolute date/time string to an arrow.Arrow object.
+
+        Try to interpret the date string as a relative time. If conversion fails, try to interpret it as an absolute
+        time. Tokens that are not recognised are returned along with the part of the string that was successfully
+        converted to an arrow object. If the date string cannot be parsed, BadArgument is raised.
+        """
+        try:
+            return arrow.utcnow().dehumanize(argument)
+        except ValueError:
+            try:
+                dt, ignored_tokens = parser.parse(argument, fuzzy_with_tokens=True)
+                return arrow.get(dt), ignored_tokens
+            except parser.ParserError:
+                raise commands.BadArgument(f"`{argument}` Could not be parsed to a relative or absolute date")
+
+
+class Epoch(commands.Cog):
+    """Convert an entered time and date to a unix timestamp."""
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: DateString = None) -> None:
+        """
+        Convert an entered date/time string to the equivalent epoch.
+
+        **Relative time**
+            accepted units: "seconds", "minutes", "hours", "days", "weeks", "months", "years"
+            Eg ".epoch in a month 4 days and 2 hours"
+
+        **Absolute time**
+            eg ".epoch 2022/6/15 16:43 -04:00"
+            Absolute times must be entered in descending orders of magnitude.
+            Timezones may take the following forms:
+                Z (UTC)
+                ±HH:MM
+                ±HHMM
+                ±HH
+
+        Times in the dropdown are shown in UTC
+        """
+        if not date_time:
```

Comment:
I thought it would be helpful to bring up the help embed if no date string was given

Reply:
I agree we should show the help embed every time there is an invalid input error, like how python does, I could probably open a separate issue to mention that.

Reply:
I prefer it in the current implementation, where if I just invoke the command with no datetime if provides the help command without an error embed.

Reply:
Which makes sense, sure, but it should be implemented on the error handler rather than on every individual command.

Bot already has it in the error handler, so we should make a issue/pull to migrate that feature over here.